ruma-events-macros/src/event_enum.rs

```diff
@@ -8,79 +8,72 @@ use syn::{
     Attribute, Expr, ExprLit, Ident, Lit, LitStr, Token,
 };
 
-use crate::event_names::{
-    ANY_EPHEMERAL_EVENT, ANY_MESSAGE_EVENT, ANY_STATE_EVENT, ANY_STRIPPED_STATE_EVENT,
-    ANY_SYNC_MESSAGE_EVENT, ANY_SYNC_STATE_EVENT, ANY_TO_DEVICE_EVENT, REDACTED_MESSAGE_EVENT,
-    REDACTED_STATE_EVENT, REDACTED_STRIPPED_STATE_EVENT, REDACTED_SYNC_MESSAGE_EVENT,
-    REDACTED_SYNC_STATE_EVENT,
-};
+fn room_event_kind(name: &EventKind, var: &EventKindVariation) -> bool {
+    matches!(
+        (name, var),
+        (EventKind::Message(_),EventKindVariation::Full)
+            | (EventKind::Message(_), EventKindVariation::Stub)
+            | (EventKind::Message(_), EventKindVariation::Redacted)
+            | (EventKind::Message(_), EventKindVariation::RedactedStub)
+            | (EventKind::State(_), EventKindVariation::Full)
+            | (EventKind::State(_), EventKindVariation::Stub)
+            | (EventKind::State(_), EventKindVariation::Redacted)
+            | (EventKind::State(_), EventKindVariation::RedactedStub)
+    )
```

Comment:
This is just `matches!(name, EventKind::Message(_) | EventKind::State(_)) && !matches!(var, EventKindVariation::Stripped | EventKindVariation::RedactedStripped)`, right?